In [3]:
import numpy as np
from matplotlib import pyplot as plt
import scipy.integrate as integrate 
from scipy.optimize import curve_fit

In [4]:
def sl(r, zeta):  # --слэйтеровская базисная функция 1s
    return np.sqrt(zeta**3/np.pi)*np.exp(-zeta*r)

def sl_int(r, zeta):   # -- для интеграла слэйтеровской базисной функции
    return (4*np.pi*r*r)*sl(r, zeta)*sl(r, zeta)
    
def gg(r, alpha):    # --гауссова базисная функция 1s
    return ((2*alpha/np.pi)**0.75)*np.exp(-alpha*r**2)

def gg_int(r, alpha):    # -- для интеграла гауссовой базисной функции
    return (4*np.pi*r*r)*gg(r, alpha)*gg(r, alpha)

In [5]:
print(integrate.quad(sl_int, 0, np.inf, args=(0.76,)))
print(integrate.quad(sl_int, 0, np.inf, args=(1.28,)))
#Проверка нормировки

(1.0000000000000002, 2.5755063924516105e-10)
(1.0000000000000009, 7.071929371832038e-09)


In [ ]:
#Определяем функцию для двух гауссианов
def two_gaussians(r, alpha1, alpha2, c1, c2):
    return (c1*gg(r, alpha1) +
            c2*gg(r, alpha2))

In [ ]:
#Блок аппроксимации функции с zeta = 0,76. Фит по ста точкам, начинаем не с нуля, так как там всегда проблемы между гауссом и слейтером, 
x = np.linspace(0.1, 4.0, 100)
zeta = 0.76
y = sl(x, zeta)
#в аргументах в curve_fit говорим, что фиттим функцию двух гауссианов, даем ось координат икс, функцию на которую фиттим(слейтер) и задаем нижние и верхние пределы значений для схождения расчета
fit_076, covar1 = curve_fit(two_gaussians, x, y, bounds=([0.01, 0.01, 0.01, 0.01],
        [40.0, 40.0, 1.0, 1.0]))

print(fit_076)


[0.16585627 1.78151228 0.90362374 0.14712999]


In [ ]:
#Блок аппроксимации функции с zeta = 1,28
zeta = 1.28
y = sl(x, zeta)

fit_128, covar2 = curve_fit(
    two_gaussians,
    x,
    y,
    bounds=(
        [0.01, 0.01, 0.01, 0.01],
        [40.0, 40.0, 1.0, 1.0]
    )
)

print(fit_128)

[0.4030094  3.83221731 0.89079471 0.19200163]


Значит наш базис выглядит так: 

H     0

S     2   1.00


      0.16585627        0.90362374
      1.78151228        0.14712999
S     2   1.00


      0.4030094        0.89079471 
      3.83221731       0.19200163



| **Basis** | **Energy, a.u.** | **SCF cycles** |
|-----------|------------|----------------|
| My Basis  | -1.10484   | 4               
| STO-3G    | -1.11615   | 1                
| 6-311G    | -1.12781   | 5              